### ISSUE TITLE
Use with CommonJS require?

### ISSUE THREAD

**Issue by a47ae** [toomuchdesign/offside#7]
Hey,
really liking your library, but is there a way to use it with browserify?
If I try 

``` javascript
var offside = require('offside-js');
offside( '#main-navigation', {
        slidingElementsSelector:'#container',
        buttonsSelector: '.toggle-navigation',
    });
```

Chrome throws `Uncaught TypeError: Cannot read property 'factory' of undefined`.

**Comment by a47ae** [comment]
Okay, solved it with

``` javascript
const offside = require('offside-js');
global.offside = offside;
offside( '#main-navigation', {
        slidingElementsSelector:'#container',
        buttonsSelector: '.toggle-navigation',
    });
```

as far as I understood just requiring offside-js doesn't binds it to window so one has to do this manually.
But maybe there exists a better solution?

**Comment by toomuchdesign** [comment]
Hi Robin, I'll be honest, I've never tested Offside as Browserify dependency. But I suspected something could have been wrong. 

I'll check it out as soon as possible. Thank you for your advice. 

Greetings!

**Comment by toomuchdesign** [comment]
Should be fixed by 0dc99c4.
I'm not happy with the patch because currently Offside still relies on window object, even when required as a CommonJS dependency.

A better solution would involve non backwards-compatible changes to plugin interface. I'll keep them for next major release (coming soon 😸).

Please let me know if 1.2.4 release solves your bug. Cheers!

**Comment by stevebauman** [comment]
Has this been fixed in v1.3? Still having the same issue :(

> TypeError: Cannot read property 'factory' of undefined

Usage:

```js
let offside = require('offside-js');

let menu = offside('#menu', {
    slidingElementsSelector: '#content',
    buttonSelector: '#menu-button'
});
```

**Comment by toomuchdesign** [comment]
Hi @stevebauman, thank you for your report.

I just pushed [a branch](https://github.com/toomuchdesign/offside/tree/fix/commonjs-import) to address this bug. Would you be able to check out if it works on your project?

I'll set up a test environment as soon as I have time, too. Sadly I had written no unit test for this plugin 😄.

**Comment by toomuchdesign** [comment]
Fixed by [v1.4.0](https://github.com/toomuchdesign/offside/releases/tag/v1.4.0).

**Comment by stevebauman** [comment]
Thanks @toomuchdesign! Appreciate it.
